Source/WebKit/NetworkProcess/cache/NetworkCacheDataCurl.cpp

```diff
@@ -113,12 +113,13 @@ RefPtr<SharedMemory> Data::tryCreateSharedMemory() const
     if (isNull() || !isMap())
         return nullptr;
 
-    HANDLE handle = std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping();
-    HANDLE newHandle;
-    if (!DuplicateHandle(GetCurrentProcess(), handle, GetCurrentProcess(), &newHandle, 0, false, DUPLICATE_SAME_ACCESS))
-        return nullptr;
+    auto fileHandle = Win32Handle::adopt(std::get<FileSystem::MappedFileData>(*m_buffer).fileMapping());
+    auto dupHandle = fileHandle.copy();
+    auto handle = SharedMemory::Handle { WTFMove(dupHandle), m_size };
+
+    (void) fileHandle.leak();
```

Comment:
Looks pretty bad. m_fileMapping of MappedFileData should be Win32Handle. Then, we can simply copy the Win32Handle.

Reply:
👍